### ISSUE TITLE
How to set a plane wave initial condition?

### ISSUE THREAD

**Issue by ZhigangPu** [NanoComp/meep#2867]
Dear community,

I'm seeking assistance with setting an initial condition for a travelling plane wave in pymeep. While the tutorials extensively cover the usage of the Source class, I'm specifically interested in directly initializing the E and B fields within the simulation.

The documentation seems to focus primarily on source-based field generation, but I'm looking for a method to explicitly set the electromagnetic field values at the simulation's start. Is there a straightforward way to achieve this in pymeep?

I would greatly appreciate any guidance on how to directly manipulate the E and B fields to create an initial travelling plane wave condition. Perhaps there's a function or method I've overlooked?
Thank you in advance for your help and expertise.

Best regards,
Ayron

**Comment by stevengj** [comment]
> I'm looking for a method to explicitly set the electromagnetic field values at the simulation's start. Is there a straightforward way to achieve this in pymeep?

Yes, you can use the `initialize_field` method of the simulation class, as [described in the manual](https://meep.readthedocs.io/en/master/Python_User_Interface/#field-computations).

> I would greatly appreciate any guidance on how to directly manipulate the E and B fields to create an initial travelling plane wave condition.

Is your whole computational cell filled with a homogeneous material?  

* If so, why are you simulating it, since you have an analytical solution?

* If not, then a planewave by itself is not a solution of Maxwell's equation, so it's not clear to me what you mean by this "initial condition".   If you want an *incident* planewave, the cleanest way to do this is with a current source (as in many examples of the docs).
